app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -6,6 +6,8 @@ module TwoFactorAuthenticatable
     before_action :handle_two_factor_authentication
     before_action :check_already_authenticated
     before_action :reset_attempt_count_if_user_no_longer_locked_out, only: :create
+
+    before_action :apply_secure_headers_override, only: :show
   end
```

Comment:
So what I mean is, the consuming controller is "too late" as far as CSP is concerned. I would love to not have this here, but am not sure what a better place is.